### PR TITLE
Graphql eslint rules added

### PR DESCRIPTION
**Pull request by vejrj**

### CODE REVIEW

File: packages/graphql-eslint-rules/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Graphql eslint rules
```
Comment (alloy) [inline packages/graphql-eslint-rules/README.md:1]:
```suggestion
# GraphQL eslint rules
```

File: packages/graphql-eslint-rules/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+This package contains eslint rules listed below
+
+- missing-apollo-key-fields
```
Comment (alloy) [inline packages/graphql-eslint-rules/README.md:5]: Can you add a short description of what it does?

File: packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+            type: "string",
+            default: DEFAULT_KEY_FIELD_NAME,
+          },
+        },
+      },
+    },
+  },
+  create(context: any) {
+    return {
+      SelectionSet(node: any) {
```
Comment (alloy) [inline packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts:155]:
```suggestion
      SelectionSet(node: SelectionSetNode) {
```

File: packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+}
+
+function getKeyFieldsObjectForCheck(keyFields: string[]) {
+  return keyFields.reduce((acc, id) => {
+    acc[id] = false;
+    return acc;
+  }, {} as Record<string, boolean>);
+}
+
+function getUnusedKeyFields(idsObjectForCheck: Record<string, boolean>) {
```
Comment (alloy) [inline packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts:80]:
```suggestion
function getUnusedKeyFields(keyFieldsObjectForCheck: Record<string, boolean>) {
```

File: packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+      return acc;
+    }
+    acc.push(key);
+    return acc;
+  }, [] as string[]);
+}
+
+export default {
+  meta: {
+    type: "suggestion",
```
Comment (alloy) [inline packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts:92]: What does this signify?

File: packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+                id
+                name
+              }
+            }
+          `,
+        },
+      ],
+    },
+    messages: {
+      [REQUIRE_KEY_FIELDS_WHEN_AVAILABLE]: `Field "{{ fieldName }}" must be selected when it's available on a type. Please make sure to include it in your selection set!\nIf you are using fragments, make sure that all used fragments {{checkedFragments}} specifies the field "{{ fieldName }}".`,
```
Comment (alloy) [inline packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts:135]: What does the output look like when there are multiple offences in a file?

**Review by alloy — COMMENTED** [review]
I would have preferred to see 2 commits, one that imports the upstream rule as-is (including link to original source) and a second commit that adds what we need.

File: packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts (hunk 5 above)
Comment (vejrj) [reply packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts]: type of the rule ... "problem" is probably better as long as we want to enforce it.

File: packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts (hunk 3 above)
Comment (vejrj) [reply packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts:155]: Doesn't work. I think the graphql-eslint-plugin enhance the original graphql `SelectionSetNode`

File: packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts (hunk 6 above)
Comment (vejrj) [reply packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts]: Changed to ```Field(s) "{{ fieldName }}" must be selected (when available on a type). Please make sure to include it in your selection set!                                                                                                     
  If you are using fragments, make sure that all used fragments {{checkedFragments}} specifies the field(s) "{{ fieldName }}".```

it returns founded unusedKeyFields 
```               
data: {
       checkedFragments: checkedFragmentSpreads.size === 0 ? "" : `(${Array.from(checkedFragmentSpreads).join(", ")})`,
       fieldName: unusedKeyFields.join(", ")
}
```

File: packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts (hunk 5 above)
Comment (alloy) [reply packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts]: Yes, ours is not a suggestion

**Comment by alloy** [comment]
:shipit:

**Pull request merged**